### ISSUE TITLE
ItemLayout inline styles not merged with theme styles

### ISSUE THREAD

**Issue by miroslavstastny** [microsoft/fluent-ui-react#467]
### Steps
```jsx
import React from 'react'
import { List, Provider } from '@stardust-ui/react'

const items = [
  {
    key: 'blue',
    content: 'Blue underline',
    styles: {
      borderBottom: '2px solid blue',
    }
  },
  {
    key: 'red',
    content: 'Red underline',
  },
]

const ListExampleSelection = () => (
  <Provider theme={{
    componentStyles: {
      ItemLayout: {
        root: {
          borderBottom: '2px solid red',
        },
      },
    },
  }}>
    <List items={items}/>
  </Provider>)

export default ListExampleSelection
```

### Expected Result
First item has a blue underline, second item has a red underline.
It can never happen that two CSS-in-JS atomic classes are applied on a single element.

### Actual Result
Two CSS classes for borderBottom are applied to the list item.
Visual results are nondeterministic as it depends on the order in which they are applied. On a fresh load both items have red underline.  💣

![image](https://user-images.githubusercontent.com/9615899/48400378-7b683300-e726-11e8-8fc2-1a77a36f8350.png)

Reported by @kohlikohl

### Version
0.11.0

### Testcase
https://codesandbox.io/s/wyj65z8pyw

**Comment by miroslavstastny** [comment]
This is actually a more general problem with how we apply CSS-in-JS classes to elements.
**The basic rule is that we cannot apply two different `renderRule` results to a single object.**
If a component in its `renderComponent` function takes `classes` from the config and passes it to another stardust component, styles are broken with non-deterministic results.

That's what we do in `ItemLayout` (and maybe other places as well).

We can fix that, but the same problem arises when a consumer uses `as` parameter for a component and passes another stardust component to it.

To play with that, you can use following testcase: https://codesandbox.io/s/9nqlpwo8y

1. The test case loads with custom styles for `MenuItem` and `Button` commented out (ignore wrong height of the `Button`, that's not related)
![image](https://user-images.githubusercontent.com/9615899/48407164-7f03b600-e736-11e8-947f-869bbc8819ec.png)
2. Uncomment `MenuItem` styles first, see the rendered result (ok)
![image](https://user-images.githubusercontent.com/9615899/48407240-a35f9280-e736-11e8-8db2-2cbb52bf3c93.png)
3. Uncomment `Button` styles, see the rendered result (ok)
![image](https://user-images.githubusercontent.com/9615899/48407260-ad819100-e736-11e8-8b75-5a4f1cb5a1b7.png)
4. Refresh the inner browser, see the rendered result (💣)
![image](https://user-images.githubusercontent.com/9615899/48407282-b96d5300-e736-11e8-9fb0-f63706d3cfd0.png)

Now start over again, but do the steps 2 and 3 in opposite order:
1. both commented out
![image](https://user-images.githubusercontent.com/9615899/48407313-cd18b980-e736-11e8-9b89-bf719893880d.png)
2. `Button` styles uncommented (ok)
![image](https://user-images.githubusercontent.com/9615899/48407529-4f08e280-e737-11e8-8eb1-90c571480a3f.png)
3. `MenuItem` styles uncommented (💣 - expected to be the same as the previous use case after step 3)
![image](https://user-images.githubusercontent.com/9615899/48407463-24b72500-e737-11e8-9d2d-3769e2fd69c4.png)
4. Refresh (💣)
![image](https://user-images.githubusercontent.com/9615899/48407494-38628b80-e737-11e8-826a-577d3b1f98c6.png)

**Comment by levithomason** [comment]
Solved in #525, we're removing the layout and item layout until we find a correct abstraction.
